File: frontend/src/core/services/game/card.ts
export enum CardColor {
  none = 0,
  red = 1,
  green = 2,
  yellow = 3,
  blue = 4,
  black = 5
}

export enum CardType {
  none = 0,
  zero = 1,
  one = 2,
  two = 3,
  three = 4,
  four = 5,
  five = 6,
  six = 7,
  seven = 8,
  eight = 9,
  nine = 10,
  skip = 11,
  reverse = 12,
  draw_two = 13,
  wild = 14,
  wild_four = 15
}

export type Card = {
  type: CardType;
  color: CardColor;
};

File: backend/Zwoo.GameEngine/Game/Cards/Card.cs
namespace Zwoo.GameEngine.Game.Cards;

public struct Card
{
    public CardColor Color = CardColor.None;
    public CardType Type = CardType.None;

    public Card(CardColor cardColor, CardType cardType)
    {
        Color = cardColor;
        Type = cardType;
    }

    public Card(int cardColor, int cardType)
    {
        Color = (CardColor)cardColor;
        Type = (CardType)cardType;
    }
}
